File: app/src/main/java/alexis/tvrs/rickandmortyuniverse/ui/episodes/EpisodeAdapter.java
package alexis.tvrs.rickandmortyuniverse.ui.episodes;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.LinearLayout;
import android.widget.TextView;

import java.util.ArrayList;

import alexis.tvrs.rickandmortyuniverse.R;

public class EpisodeAdapter extends BaseAdapter {

    private Context context;
    private ArrayList<Episode> list;

    EpisodeAdapter(Context context, ArrayList<Episode> list) {
        this.context = context;
        this.list = list;
    }

    @Override
    public int getCount() {
        return list.size();
    }

    @Override
    public Object getItem(int position) {
        return list.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        LinearLayout layoutItem;
        LayoutInflater mInflater = LayoutInflater.from(context);

        if (convertView == null) {
            layoutItem = (LinearLayout) mInflater.inflate(R.layout.episode_layout, parent, false);
        } else {
            layoutItem = (LinearLayout) convertView;
        }

        TextView nameView =layoutItem.findViewById(R.id.characterLayout_genre);
        String nameText = "Name: "+list.get(position).getName();
        nameView.setText(nameText);

        TextView orderView =layoutItem.findViewById(R.id.characterLayout_status);
        String orderText = "Order: "+list.get(position).getEpisodeOrder();
        orderView.setText(orderText);

        TextView airDateView =layoutItem.findViewById(R.id.characterLayout_species);
        String airDateText = "Air Date: "+list.get(position).getAir_date();
        airDateView.setText(airDateText);


        return layoutItem;
    }
}

File: app/src/main/java/alexis/tvrs/rickandmortyuniverse/wiki/adapters/EpisodeAdapter.kt
package alexis.tvrs.rickandmortyuniverse.wiki.adapters

import alexis.tvrs.rickandmortyuniverse.R
import alexis.tvrs.rickandmortyuniverse.wiki.models.Episode
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.LinearLayout
import kotlinx.android.synthetic.main.adapter_episode.view.*
import java.util.*

class EpisodeAdapter internal constructor(private val context: Context, private val list: ArrayList<Episode>) : BaseAdapter() {
    override fun getCount(): Int {
        return list.size
    }

    override fun getItem(position: Int): Any {
        return list[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val layoutItem: LinearLayout
        val mInflater = LayoutInflater.from(context)
        layoutItem = if (convertView == null) {
            mInflater.inflate(R.layout.adapter_episode, parent, false) as LinearLayout
        } else {
            convertView as LinearLayout
        }

        val nameText = "Name: " + list[position].name
        layoutItem.episode_layout_name.text = nameText

        val orderText = "Order: " + list[position].episode
        layoutItem.episode_layout_order.text = orderText

        val airDateText = "Air Date: " + list[position].airDate
        layoutItem.episode_layout_air_date.text = airDateText

        return layoutItem
    }
}